pennylane/gradients/parameter_shift_cv.py

```diff
@@ -0,0 +1,678 @@
+        function to be applied to the evaluated tapes.
+    """
+    argnum = argnum or tape.trainable_params
+
+    # Determine the locations of any variance measurements in the measurement queue.
+    var_mask = [m.return_type is qml.operation.Variance for m in tape.measurements]
+    var_idx = np.where(var_mask)[0]
+
+    # Get <A>, the expectation value of the tape with unshifted parameters.
+    expval_tape = tape.copy(copy_operations=True)
+
+    # Convert all variance measurements on the tape into expectation values
+    for i in var_idx:
+        obs = expval_tape._measurements[i].obs
+        expval_tape._measurements[i] = qml.measure.MeasurementProcess(
+            qml.operation.Expectation, obs=obs
+        )
+
+    gradient_tapes = [expval_tape]
+
+    # evaluate the analytic derivative of <A>
+    pdA_tapes, pdA_fn = expval_param_shift(expval_tape, argnum, shift, gradient_recipes, f0)
+    gradient_tapes.extend(pdA_tapes)
+
+    # Store the number of first derivative tapes, so that we know
+    # the number of results to post-process later.
+    tape_boundary = len(pdA_tapes) + 1
+    expval_sq_tape = tape.copy(copy_operations=True)
+
+    for i in var_idx:
+        # We need to calculate d<A^2>/dp; to do so, we replace the
+        # observables A in the queue with A^2.
+        obs = expval_sq_tape._measurements[i].obs
+
+        # CV first order observable
+        # get the heisenberg representation
+        # This will be a real 1D vector representing the
+        # first order observable in the basis [I, x, p]
+        A = obs._heisenberg_rep(obs.parameters)
+
+        # take the outer product of the heisenberg representation
+        # with itself, to get a square symmetric matrix representing
+        # the square of the observable
+        obs = qml.PolyXP(np.outer(A, A), wires=obs.wires)
+        expval_sq_tape._measurements[i] = qml.measure.MeasurementProcess(
+            qml.operation.Expectation, obs=obs
+        )
+
+    # Non-involutory observables are present; the partial derivative of <A^2>
+    # may be non-zero. Here, we calculate the analytic derivatives of the <A^2>
+    # observables.
+    pdA2_tapes, pdA2_fn = second_order_param_shift(
+        expval_sq_tape, dev_wires, argnum, shift, gradient_recipes
+    )
+    gradient_tapes.extend(pdA2_tapes)
+
+    def processing_fn(results):
+        mask = qml.math.convert_like(qml.math.reshape(var_mask, [-1, 1]), results[0])
+        f0 = qml.math.expand_dims(results[0], -1)
+
+        pdA = pdA_fn(results[1:tape_boundary])
+        pdA2 = pdA2_fn(results[tape_boundary:])
+
+        # return d(var(A))/dp = d<A^2>/dp -2 * <A> * d<A>/dp for the variances (mask==True)
+        # d<A>/dp for plain expectations (mask==False)
+        return qml.math.where(mask, pdA2 - 2 * f0 * pdA, pdA)
+
+    return gradient_tapes, processing_fn
+
+
+def second_order_param_shift(tape, dev_wires, argnum=None, shift=np.pi / 2, gradient_recipes=None):

```

Comment:
This method is ported from `CVParamShiftTape.parameter_shift_second_order`, with some modifications (listed below).